core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -1126,8 +1128,12 @@ class HttpGoogleServicesDAO(
     cloudResourceManager
   }
 
-  private def getUserCredential(userInfo: UserInfo): Credential = {
-    new GoogleCredential().setAccessToken(userInfo.accessToken.token).setExpiresInSeconds(userInfo.accessTokenExpiresIn)
+  private def getUserCredential(userInfo: UserInfo): Option[Credential] = {
+    // Use the Google token if present to build the credential
+    val tokenOpt = if (userInfo.isB2C) userInfo.googleAccessTokenThroughB2C else Some(userInfo.accessToken)
+    tokenOpt.map { googleToken =>
+      new GoogleCredential().setAccessToken(googleToken.token).setExpiresInSeconds(userInfo.accessTokenExpiresIn)
```

Comment:
Btw it was bothering me that this class uses `GoogleCredential` (which is deprecated) and pem files instead of JSON keys for service accounts. I started to refactor how Google clients/credentials are created but it turned into a bigger change so I backed it out of this PR. May PR that separately if I finish. ;)